sympy/stats/frv_types.py

```diff
@@ -136,14 +144,21 @@ def check(sides):
         _value_check((sides.is_positive, sides.is_integer),
                     "number of sides must be a positive integer.")
 
+    @property
+    def is_symbolic(self):
+        return _is_sym_dim(self.sides)
+
     @property
     @cacheit
     def dict(self):
-        as_int(self.sides) # Check that self.sides can be converted to an integer
+        if self.is_symbolic:
+            return Density(self)
```

Comment:
Need to find a way to put this in the superclass, we cannot replicate this in all classes.